src/PIL/_util.py

```diff
@@ -4,10 +4,10 @@
 from pathlib import Path
 from typing import Any, NoReturn
 
-from ._typing import TypeGuard
+from ._typing import StrOrBytesPath, TypeGuard
 
 
-def is_path(f: Any) -> TypeGuard[bytes | str | Path]:
+def is_path(f: Any) -> TypeGuard[StrOrBytesPath]:
```

Comment:
Given that the function returns `isinstance(f, (bytes, str, Path))`, I'm not sure I understand the point of this change.
I haven't checked, but I think this will reintroduce a type warning in `ImageFont` here: https://github.com/python-pillow/Pillow/blob/b3a7ae065c4f34b345ecaa3b019bbd1d24e7922c/src/PIL/ImageFont.py#L232-L236

Reply:
Good point — I think I gave @hugovk bad advice here.

I would probably consider refactoring the code you link to there in ImageFont.py so that it does handle any `os.PathLike` rather than specifically checking for `pathlib.Path`, and change this function so that it does `isinstance(f, (str, bytes, os.PathLike))` instead of `isinstance(f, (str, bytes, pathlib.Path))`. But perhaps all that could be deferred to another PR

Reply:
> I haven't checked, but I think this will reintroduce a type warning in `ImageFont` here:

We haven't added type hints to `ImageFont.py` yet, it's giving 55 strict warnings for both `main` and this PR, and zero non-strict on both.


> I would probably consider refactoring the code you link to there in ImageFont.py so that it does handle any `os.PathLike` rather than specifically checking for `pathlib.Path`, and change this function so that it does `isinstance(f, (str, bytes, os.PathLike))` instead of `isinstance(f, (str, bytes, pathlib.Path))`. But perhaps all that could be deferred to another PR

I've updated this PR to include the refactor.